src/libraries/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.Windows.cs

```diff
@@ -176,7 +176,7 @@ private static void CreateCSP(CspParameters parameters, bool randomKeyContainer,
         internal static void AcquireCsp(CspParameters cspParameters, out SafeProvHandle safeProvHandle)
         {
             Debug.Assert(cspParameters != null);
-            Debug.Assert(cspParameters.KeyContainerName == null);
+            Debug.Assert(OperatingSystem.IsWindows() && cspParameters.KeyContainerName == null);
```

Comment:
Separate asserts, please.

Reply:
The warning is caught within the Assert, `cspParameters.KeyContainerName` is windows only

Reply:
Hm.  Alright.  I'd still rather it was separate asserts, but hopefully if someone ever violates the call contract (and calls the helper method with a null KeyContainerName) they realize "this code is in a file that says `.Windows` in the name, it's probably not the left half of this expression".

Reply:
OK, good point